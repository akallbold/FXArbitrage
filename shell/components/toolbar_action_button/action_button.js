import React from 'react'
import { connect } from 'react-redux'
import { get } from 'lodash'
import { DropdownButton, Button } from 'playa-styleguide'
import { push } from 'redux-little-router'
import { curChannelUrlSelector, curChannelSelector } from 'selectors/platform_selectors'

const dropdownOptions = [{ name: 'Bulk Upload', id: 'bulkUpload' }]

export const ActionButtonComponent = ({ channelUrl, channel, pushRoute }) => {
  const onButtonClick = () => (
    pushRoute(`${channelUrl}/manager/campaigns/new`)
  )

  const onOptionClick = (e) => {
    if (get(e, 'target.value') !== 'bulkUpload') return
    pushRoute(`${channelUrl}/bulk-upload`)
  }

  if (channel === 'twitter') {
    return (
      <Button
        onClick={onButtonClick}
        size='large'
      >
        Create Campaign
      </Button>
    )
  }

  return (
    <DropdownButton
      onClick={onButtonClick}
      onOptionClick={onOptionClick}
      listData={dropdownOptions}
      listType='standardDropdown'
      lockBody
    >
      Create Campaign
    </DropdownButton>
  )
}

const mapDispatchToProps = { pushRoute: push }

const mapStateToProps = state => ({
  channelUrl: curChannelUrlSelector(state),
  channel: curChannelSelector(state)
})

export default connect(mapStateToProps, mapDispatchToProps)(ActionButtonComponent)
