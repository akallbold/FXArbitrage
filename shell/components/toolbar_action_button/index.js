export ActionButton from './action_button'
