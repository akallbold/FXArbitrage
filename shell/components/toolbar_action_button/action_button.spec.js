import React from 'react'
import { ActionButtonComponent } from 'toolbar_action_button/action_button'
import { shallow } from 'enzyme'

describe('render', () => {
  it('renders the component as expected', () => {
    const wrapper = shallow(<ActionButtonComponent />)

    expect(wrapper).toMatchSnapshot()
  })
})
