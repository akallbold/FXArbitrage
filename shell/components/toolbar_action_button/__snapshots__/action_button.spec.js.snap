// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders the component as expected 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <DropdownButton
    listData={
      Array [
        Object {
          "id": "bulkUpload",
          "name": "Bulk Upload",
        },
      ]
    }
    listType="standardDropdown"
    lockBody={true}
    onClick={[Function]}
    onOptionClick={[Function]}
  >
    Create Campaign
  </DropdownButton>,
  "nodes": Array [
    <DropdownButton
      listData={
        Array [
          Object {
            "id": "bulkUpload",
            "name": "Bulk Upload",
          },
        ]
      }
      listType="standardDropdown"
      lockBody={true}
      onClick={[Function]}
      onOptionClick={[Function]}
    >
      Create Campaign
    </DropdownButton>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ActionButtonComponent />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <DropdownButton
          listData={
            Array [
              Object {
                "id": "bulkUpload",
                "name": "Bulk Upload",
              },
            ]
          }
          listType="standardDropdown"
          lockBody={true}
          onClick={[Function]}
          onOptionClick={[Function]}
        >
          Create Campaign
        </DropdownButton>,
        "_debugID": 2,
        "_renderedOutput": <DropdownButton
          listData={
            Array [
              Object {
                "id": "bulkUpload",
                "name": "Bulk Upload",
              },
            ]
          }
          listType="standardDropdown"
          lockBody={true}
          onClick={[Function]}
          onOptionClick={[Function]}
        >
          Create Campaign
        </DropdownButton>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ActionButtonComponent />,
}
`;
